ksql-engine/src/main/java/io/confluent/ksql/function/udaf/KudafInitializer.java

```diff
@@ -21,19 +21,30 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 public class KudafInitializer implements Initializer<GenericRow> {
 
-  private final List initialGenericRowColumns;
+  private final List<Supplier> aggValueSuppliers = new ArrayList<>();
+  private final int nonAggValSize;
 
-  public KudafInitializer(List initialGenericRowColumns) {
-    this.initialGenericRowColumns = initialGenericRowColumns;
+  public KudafInitializer(final int nonAggValSize) {
+    this.nonAggValSize = nonAggValSize;
   }
 
   @Override
   public GenericRow apply() {
-    List rowColumns = new ArrayList(initialGenericRowColumns);
-    return new GenericRow(rowColumns);
+    final List<Object> values = IntStream.range(0, nonAggValSize)
+        .mapToObj(value -> null)
```

Comment:
Not sure I follow this line. Will `values` just be a list of `null` of size `nonAggValSize`?

Reply:
Yes, it needs to be a list of size `numberOfColumns` of empty/default values. This hasn't really changed, i just moved the code here as the list needs to be new instance each time this is called so values are re-initialized, i.e., reference types can't be shared across initilialization